### ISSUE TITLE
Issues with Kindle android app.

### ISSUE THREAD

**Issue by Tikotik** [Ephemerality/xray-builder.gui#31]
Versions: 
X-ray 2.1.24, Calibre 3.31, Kindle android app 8.11.0.57

What I did: 
fixed asin - B005FVNIW6 from calibre plugin
used android output to build xray files, from [shelfari archive](https://web.archive.org/web/20131227102953/http://www.shelfari.com/books/35825/Enders-Game) for [this](https://github.com/tmilovanov/wisecreator/blob/master/test%20books/Orson%20Scott%20Card%20-%20Ender's%20Game.mobi) test file.
copied the ASIN-fixed and saved .mobi book and xray o/p folder to `sdcard/Android/data/com.amazon.kindle/files/`

Result:
x-ray button and author profile does not show up in app.
But the author profile and X-ray o/p files are good in general, and new amazon cover for the book is downloaded automatically by app based on ASIN.

<sub>I posted multiple times on mobileread support thread, but moderators are not approving my comment, Idk whats wrong in this?</sub>

<details>
<summary>Click to expand logs</summary>
Running X-Ray Builder GUI v2.1.24.0. Log started on 04-11-2018 at 14:24:40.
Extracting metadata...
Got metadata!
Database Name: Ender_8217_s_Game
UniqueID: 3138608418
Extracting metadata...
Got metadata!
Database Name: Ender_8217_s_Game
UniqueID: 3138608418
Invalid X-Ray file.
Extracting metadata...
Saving rawML to dmp directory...
Got metadata!
Database Name: Ender_8217_s_Game
UniqueID: 3138608418
Book's Shelfari URL: https://web.archive.org/web/20131227102953/http://www.shelfari.com/books/35825/Enders-Game
Attempting to build Author Profile...
Searching for author's page on Amazon.com...
Author page found on Amazon!
Author's Amazon Page URL: https://www.amazon.com/Orson-Scott-Card/e/B000AQ3SS0
Author biography found on Amazon!
Saving biography to G:\xray-builder-gui-v2.1.24\ext\B000AQ3SS0.bio
Using biography from G:\xray-builder-gui-v2.1.24\ext\B000AQ3SS0.bio.
Downloading author image...
Grayscale base64-encoded author image created!
Gathering author's other books...
Gathering metadata for other books...
Writing Author Profile to file...
Author Profile file created successfully!
Saved to G:\xray-builder-gui-v2.1.24\out\Android\B005FVNIW6\AuthorProfile.profile.B005FVNIW6.asc
Attempting to build Start Actions and End Actions...
Attempting to find book on Amazon...
Book found on Amazon!
Gathering recommended book metadata...
An error occurred finding next book in series: The method or operation is not implemented.
   at XRayBuilderGUI.DataSources.Secondary.Shelfari.GetExtrasAsync(BookInfo curBook, IProgressBar progress, CancellationToken cancellationToken) in C:\projects\xray-builder-gui\src\DataSources\Secondary\Shelfari.cs:line 108
   at XRayBuilderGUI.EndActions.<GenerateNewFormatData>d__15.MoveNext() in C:\projects\xray-builder-gui\src\EndActions.cs:line 337
An error occurred while creating the new Author Profile, Start Actions, and/or End Actions files:
The method or operation is not implemented.
   at XRayBuilderGUI.DataSources.Secondary.Shelfari.GetExtrasAsync(BookInfo curBook, IProgressBar progress, CancellationToken cancellationToken) in C:\projects\xray-builder-gui\src\DataSources\Secondary\Shelfari.cs:line 108
   at XRayBuilderGUI.EndActions.<GenerateNewFormatData>d__15.MoveNext() in C:\projects\xray-builder-gui\src\EndActions.cs:line 354
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at XRayBuilderGUI.UI.frmMain.<btnKindleExtras_Run>d__25.MoveNext() in C:\projects\xray-builder-gui\src\UI\frmMain.cs:line 432
Extracting metadata...
Saving rawML to dmp directory...
Got metadata!
Database Name: Ender_8217_s_Game
UniqueID: 3138608418
Book's Shelfari URL: https://web.archive.org/web/20131227102953/http://www.shelfari.com/books/35825/Enders-Game
Attempting to build X-Ray...
Offset: 0
Downloading Shelfari page...
Duplicate term "Andrew "Ender" Wiggin" found. Ignoring this duplicate.
Duplicate term "Alai" found. Ignoring this duplicate.
Duplicate term "Polemarch" found. Ignoring this duplicate.
Downloading notable clips...
59 Terms found on Shelfari:
Andrew Wiggin (aka Ender), Valentine Wiggin, Peter Wiggin, Bean, Colonel Hyrum Graff, Alai, Bonzo de Madrid, Mazer Rackham, Bernard, Petra Arkanian, Dink Meeker, Stilson, Carn Carby, Demosthenes, Locke, Major Anderson, Hot Soup, Mick, Miss Pumphrey, Rose the Nose, Pol Slattery, Major Imbu, Crazy Tom, Dr. Lineberry, Captain Dap, Shen, Talo Momoe, Cam Carby, William Bee, Abra, Admiral Chamrajnagar, Mother, Andrew "Ender" Wiggin, Isaac Asimov, Napoleon, Alexander (the Great), Polemarch, Dr Device, Julius Caesar, Dad, Valentine, Condor, Lee, Bruce Catton, Queen Bugger, International Fleet, The Warsaw Pact, Battle School, Eros, Ansible, Battle Room, Buggers/Formics, Hegemon, Hook, Launchies, Merda, Third, Toon, Strategos, 
Character aliases read from G:\xray-builder-gui-v2.1.24\ext\B005FVNIW6.aliases.
Initial X-Ray built, adding locations and chapters...
Chapters read from G:\xray-builder-gui-v2.1.24\ext\B005FVNIW6.chapters.
Delete this file if you want chapters built automatically.
Reloaded chapters from edited file.
Found chapters:
1. Third | start: 2332 | end: 18883
2. Peter | start: 18883 | end: 34323
3. Graff | start: 34323 | end: 59717
4. Launch | start: 59717 | end: 82701
5. Games | start: 82701 | end: 121104
6. The Giants Drink | start: 121104 | end: 147482
7. Salamander | start: 147482 | end: 218886
8. Rat | start: 218886 | end: 270234
9. Locke and Demosthenes | start: 270234 | end: 349848
10. Dragon | start: 349848 | end: 391957
11. Veni Vidi Vici | start: 391957 | end: 452715
12. Bonzo | start: 452715 | end: 515352
13. Valentine | start: 515352 | end: 579165
14. Enders Teacher | start: 579165 | end: 694977
15. Speaker for the Dead | start: 694977 | end: 742972
Scanning book content...
Scan time: 00:00:03.0593858
No locations were found for the term "Andrew Wiggin (aka Ender)".
You should add aliases for this term using the book or rawml as a reference.
No locations were found for the term "Captain Dap".
You should add aliases for this term using the book or rawml as a reference.
No locations were found for the term "Cam Carby".
You should add aliases for this term using the book or rawml as a reference.
No locations were found for the term "Isaac Asimov".
You should add aliases for this term using the book or rawml as a reference.
No locations were found for the term "Alexander (the Great)".
You should add aliases for this term using the book or rawml as a reference.
No locations were found for the term "Dr Device".
You should add aliases for this term using the book or rawml as a reference.
No locations were found for the term "Bruce Catton".
You should add aliases for this term using the book or rawml as a reference.
No locations were found for the term "Buggers/Formics".
You should add aliases for this term using the book or rawml as a reference.
No locations were found for the term "Merda".
You should add aliases for this term using the book or rawml as a reference.
Saving X-Ray to file...
Building new X-Ray database. May take a few minutes...
Done building initial database. Populating with info from source X-Ray...
Updating database with terms, descriptions, and excerpts...
Writing 59 terms...
Writing 1792 excerpts...
Adding notable clips...
Writing top mentions...
Writing metadata...
Updating indices...
X-Ray previewData file created successfully!
Saved to G:\xray-builder-gui-v2.1.24\out\Android\B005FVNIW6\XRAY.B005FVNIW6.previewData
X-Ray file created successfully!
Saved to G:\xray-builder-gui-v2.1.24\out\Android\B005FVNIW6\XRAY.B005FVNIW6.Ender_8217_s_Game_.db
</details>

**Comment by Tikotik** [comment]
Yeah, the book is at `sdcard/Android/data/com.amazon.kindle/files/`
and the x-ray is at `/sdcard/Android/data/com.amazon.kindle/files/B005FVNIW6/XRAY.ASIN.bookname.db`

Here is book and xray o/p as [Ephemeral zip](https://file.io/tqwuGm)

<sub>I cant find how to PM on mobileread.</sub>

**Comment by Ephemerality** [comment]
Ahh I see. There was a tiny bug that was causing the file to be named incorrectly. It should be `XRAY.B005FVNIW6.Ender_8217_s_Game_BB135D22.db` (the unique ID was missing from the GUID).  
Fixed in a37460f79a3fc89ff1bb92b56e39dbaf2a303102 (release 2.1.30)  
I built the Start/EndActions and they seemed to work. I don't think the AuthorProfiles actually work on Android, but I'll have to look into that more.
